### PR TITLE
Make our output JS less terrible.

### PR DESCRIPTION
**Pull request by patricksrobertson**
This reforms our template output so it's a bit more ES6-7 compatible.

### CODE REVIEW

File: examples/output/model/v3/patient.js [hunk 1]
```diff
@@ -1,37 +1,39 @@
...
-    return adapter.autocomplete(modelName, this.get('id'), arguments).then(function(response) {
-      var serializer = _this.store.serializerFor(modelName);
-      var payloadKey = serializer.payloadKeyFromModelName(modelName);
-      var payload = response[payloadKey];
+  autocomplete() {
+    let _this = this;
+    const modelName = this.constructor.modelName;
+    const adapter = this.store.adapterFor(modelName);
+    return adapter.autocomplete(modelName, this.get('id'), arguments).then((response) => {
+      const serializer = _this.store.serializerFor(modelName);
```
Comment (alexrothenberg) [inline examples/output/model/v3/patient.js]: is `_this` unnecessary now that we're using `=>` function ?

File: lib/braise/adapter_template.ex [hunk 2]
```diff
@@ -57,9 +60,9 @@ defmodule Braise.AdapterTemplate do
   def path_for_type(resource_name) do
     if String.match?(resource_name, ~r/_/) do
       """
-      pathForType: function(type) {
-          var underscorized = Ember.String.underscore(type);
-          return Ember.String.pluralize(underscorized);
+      pathForType(type) {
+          var underscorized = EmberString.underscore(type);
```
Comment (alexrothenberg) [inline lib/braise/adapter_template.ex]: `const` ?

File: lib/braise/adapter_template.ex [hunk 3]
```diff
@@ -93,7 +96,7 @@ defmodule Braise.AdapterTemplate do
     action_name = link_action.name
     method = link_action.method
     """
-    #{action_name}: function(modelName, id, snapshot) {
+    #{action_name}(modelName, id, snapshot) {
         var url = this.buildURL(modelName, id) + '/#{action_name}';
```
Comment (alexrothenberg) [inline lib/braise/adapter_template.ex]: `const` ?

File: test/adapter_template_test.exs [hunk 4]
```diff
@@ -57,12 +60,15 @@ defmodule AdapterTemplateTest do
     import DS from 'ember-data';
     import Ember from 'ember';
 
-    export default DS.RESTAdapter.extend({
+    const { RESTAdapter } = DS;
+    const { computed, EmberString: String } = Ember;
```
Comment (alexrothenberg) [inline test/adapter_template_test.exs]: I think this should be the other way

`const { computed, String: EmberString } = Ember;`

File: test/adapter_template_test.exs [hunk 5]
```diff
@@ -74,7 +80,7 @@ defmodule AdapterTemplateTest do
         return this._super(url, type, options);
       },
 
-      cancel: function(modelName, id, snapshot) {
+      cancel(modelName, id, snapshot) {
         var url = this.buildURL(modelName, id) + '/cancel';
```
Comment (alexrothenberg) [inline test/adapter_template_test.exs]: If we're ES6'ing all this should we standardize to use `let` or `const` here (& everywhere else) too?

File: examples/output/model/v3/patient.js (hunk 1 above)
Comment (patricksrobertson) [inline examples/output/model/v3/patient.js]: I'm not 100% sure the anon function works there with how Ember does things :(. Will test it out.

**Pull request merged**